dist/profile/manifests/jenkinsplugin.pp

```diff
@@ -7,12 +7,14 @@
   validate_string($name)
 
   exec { "install-plugin-${name}":
+    ## Check for plugin presence on the HOST (e.g. with the jenkins home in "/var/lib/jenkins" on the filesystem)
+    unless    => "/usr/bin/test -f /var/lib/jenkins/plugins/${name}.jpi || /usr/bin/test -f /var/lib/jenkins/plugins/${name}.hpi",
+    require   => Docker::Run['jenkins'],
```

Comment:
Can you install a plugin is Jenkins is still booting? 
I think we may face a racing condition where the container is started but the Jenkins instance inside is not ready yet.
I am wondering if we shouldn't have an additional check to be sure that Jenkins is ready

Reply:
Good observation, Let me add a comment to explain it for later readers.

In the current state, the plugin will always be able to be installed unless the container is killed:

- There are no healthchecks defined, so the status "container running" is reached as soon as the process with the PID 1 is up
- The PID 1 is [`tini`](https://github.com/jenkinsci/docker/blob/master/11/debian/buster/hotspot/Dockerfile#L105) which means that even if Jenkins is not able to start successfully (example: JCasc issues, HTTP/500, etc.) , there should be always a java process
- If you run a `docker exec` command (such as this one), then `tini` manages all the sub processes: while the plugin installation is running, then tini should not exit (unless a kill from systemd or an host admin of course)

However you are correct that if we reach a status where Jenkins is unable to expose its API to localhost (not the case with JCasc failing, but the startup sequence is not an easy one and can have edge cases), then the plugin installation would fail.

I got a proposal (but subsequent PR) that would be to edit the `idempotent-cli` script (on the host) to detect the passed command: if it is `install-plugin <...>` then it should use the [`jenkins-plugin-cli`](https://github.com/jenkinsci/docker#plugin-installation-manager-cli) in the container with the argument `--plugins-directory=/var/jenkins-home/plugins`], so there would be no dependency to Jenkins's status for the plugins installation (but we have to keep the `jenkins-cli.jar` for running groovy and triggering a `safe-restart` though). WDYT?